projects/RabbitMQ.Client/client/impl/Frame.cs

```diff
@@ -147,12 +147,12 @@ internal static class Heartbeat
                 Constants.FrameEnd
             };
 
-            public static Memory<byte> GetHeartbeatFrame()
+            public static ReadOnlyMemory<byte> GetHeartbeatFrame(ArrayPool<byte> pool)
             {
                 // Is returned by SocketFrameHandler.WriteLoop
-                var buffer = ArrayPool<byte>.Shared.Rent(FrameSize);
+                var buffer = pool.Rent(FrameSize);
```

Comment:
Where is corresponding `pool.Return` call for this rented buffer?

Reply:
In the same place as mentioned by comment (`WriteLoop` method)

Reply:
Fixed